src/formatDate.js

```diff
@@ -8,7 +8,27 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separatorFrom = fromFormat.pop();
+  const separatorTo = toFormat.pop();
+
+  const dateParts = date.split(separatorFrom);
+  const dateMap = {};
+
+  fromFormat.forEach((part, index) => {
+    dateMap[part] = dateParts[index];
+  });
+
+  const formattedDateParts = toFormat.map((part) => {
+    if (part === 'YYYY' && dateMap['YY']) {
+      return dateMap['YY'] < 30 ? '20' + dateMap['YY'] : '19' + dateMap['YY'];
+    } else if (part === 'YY' && dateMap['YYYY']) {
+      return dateMap['YYYY'].slice(-2);
```

Comment:
Similarly, this logic assumes that if 'YY' is requested and 'YYYY' is provided, it will convert 'YYYY' to 'YY'. Ensure that 'YYYY' exists in `dateMap` before applying this logic.